### ISSUE TITLE
how to use weblaf's TooltipManager on the header of a JTable

### ISSUE THREAD

**Issue by mokun** [mgarin/weblaf#471]
Currently, I have no problem using weblaf's tip on Swing JComponent such as JLabel.

```java
TooltipManager.setTooltip (label, Msg.getString("TabPanelHealth.tooltip"), TooltipWay.down);
```

Say I have a table with 4 columns and eventually I'd like to provide a tooltip text for each column as the mouse pointer hover over the column header.

How do you override JTableHeader's getToolTipText() in such a way to use weblaf's TooltipManager's setTooltip() ?

```java
		JTable radiationTable = new JTable(radiationTableModel) {
		    protected JTableHeader createDefaultTableHeader() {
		        return new JTableHeader(columnModel) {
		            public String getToolTipText(MouseEvent e) {
		                //String tip = null;
		                java.awt.Point p = e.getPoint();
		                int index = columnModel.getColumnIndexAtX(p.x);
		                if (index > -1) {
			                int realIndex = columnModel.getColumn(index).getModelIndex();
			                return radiationToolTips[realIndex];
		            	}
		                else {
		                	return "Radiation Dosage Equivalent";
		                }
		            }
		        };
		    }
```

**Comment by mgarin** [comment]
Unfortunately, because these custom tooltips are not related to default `JToolTip` component anyhow, it is not possible to force Swing to provide WebLaF tooltips instead of default ones.

There is a way to enable them for cells in list, table and tree components, but it will require usage of WebLaF components instead of default Swing ones.

Here is a table example:
```java
public class TableTooltips
{
    public static void main ( final String[] args )
    {
        SwingUtils.invokeLater ( new Runnable ()
        {
            @Override
            public void run ()
            {
                WebLookAndFeel.install ();

                final String[] columns = new String[]{ "1", "2", "3" };
                final String[][] data = new String[][]{ { "1", "2", "3" }, { "1", "2", "3" }, { "1", "2", "3" }, { "1", "2", "3" } };
                final WebTable table = new WebTable ( data, columns );
                table.setToolTipProvider ( new TableToolTipProvider ()
                {
                    @Override
                    public String getToolTipText ( final WebTable component, final Object value, final int index, final int column,
                                                   final boolean isSelected )
                    {
                        return "Cell(row,col): " + index + "," + column;
                    }
                } );

                TestFrame.show ( new JScrollPane ( table ) );
            }
        } );
    }
}
```
You can also override `TableToolTipProvider` class and customize tooltip settings to your liking.

Here is the resulting tooltip look:
![image](https://user-images.githubusercontent.com/2361389/32706031-29ab67c8-c82b-11e7-91c9-c1f3fe0fb757.png)

Although support for the table header is not there yet, but its a quite minor thing, so I will add it on this week along with some other improvements. It will basically use the same provider, but header tooltips will be requested under `-1` row index.

**Comment by mokun** [comment]
I see. Unfortunately, for most part, I can't use `WebTable` without some code restructuring on my app. 

So I'll just have to wait for the future release. Thanks and keep up the good work !

**Comment by mgarin** [comment]
I changed this issue to enhancement so I won't lose it. 

Also I'll still be going over some parts of the tooltip system before final 1.2.9 release, so I'll try to check whether there could be some simple solution or not.

**Comment by mgarin** [comment]
I will be adding improvements to the advanced tooltip providers so that you can use them with default `J`-components through client properties.

**Comment by mgarin** [comment]
Short list of planned tooltip changes: #520

**Comment by mokun** [comment]
awesome !

**Comment by mgarin** [comment]
Related improvements have been implemented and will be available in v1.2.10 update.

Shortly - now you're able to provide table header and tabbed pane custom tooltips in a similar way how it was possible in list, tree and table cells. There are some new related examples available in the the demo application.

Also all of the custom tooltip providers are now set through `JComponent` client properties, therefore they are available for common Swing components like `JList`, `JTable` etc.

Here is an example with two providers set in `JTable`:
```java
final JTable table = new JTable ( ... );
table.putClientProperty ( WebTable.HEADER_TOOLTIP_PROVIDER_PROPERTY, new TableHeaderToolTipProvider<String> ()
{
    @Nullable
    @Override
    protected String getToolTipText ( @NotNull final JTableHeader tableHeader,
                                      @NotNull final TableHeaderCellArea<String, JTableHeader> area )
    {
        return "My header tooltip";
    }
} );
table.putClientProperty ( WebTable.TOOLTIP_PROVIDER_PROPERTY, new TableToolTipProvider<Object> ()
{
    @Nullable
    @Override
    protected String getToolTipText ( @NotNull final JTable table, 
                                      @NotNull final TableCellArea<Object, JTable> area )
    {
        return "My cell tooltip";
    }
} );
```
Respective public static properties can be found in all `Web`-component versions of `J`-components.

**Comment by mgarin** [comment]
A small screenshot of the demo example regarding the table header tooltips:
![image](https://user-images.githubusercontent.com/2361389/65345563-400a8080-dbe3-11e9-8762-f7527c42c60e.png)
